lib/shared/bucketing-assembly-script/assembly/helpers/arrayHelpers.ts

```diff
@@ -0,0 +1,50 @@
+class SortingArrayItem<T> {
+    value: string
+    entry: T
+}
+
+export type SortingArray<T> = SortingArrayItem<T>[]
+
+/**
+ * Sorts an array of objects that have been preformatted to the "SortingArrayItem" format which includes the original
+ * item under the "entry" field, and the value to sort by under the "value" field. Returns a sorted array of "entries"
+ * @param arr
+ */
+export function sortObjectsByString<T>(arr: SortingArray<T>): T[] {
+    const sorted = arr.sort((a, b) => {
+        let cursor = 0
+
+        // advance cursor while characters are the same
+        while (a.value.charCodeAt(cursor) === b.value.charCodeAt(cursor)) {
+            cursor++
+        }
+
+        // check if we're out of bounds on the first string
+        if (isNaN(a.value.charCodeAt(cursor))) {
+            // check if we're out of bounds on the second string
+            if (isNaN(b.value.charCodeAt(cursor))) {
+                // strings are the same length and all characters are identical
+                return 0
+            } else {
+                // second string is longer
+                return -1
+            }
+        }
+
+
+        if (isNaN(b.value.charCodeAt(cursor))) {
+            // first string is longer
+            return 1
+        }
+
+        // return subtraction of the current cursor position's character codes (which must be different)
+        return a.value.charCodeAt(cursor) - b.value.charCodeAt(cursor)
+    })
+
+    let result: T[] = []
+    for (let i = 0; i < sorted.length; i++) {
+        result.push(sorted[i].entry)
+    }
```

Comment:
you could do a `.map()` on this if you want.

Reply:
I actually tried that but it refused to compile and complained that the types were wrong, no matter how explicitly I wrote them out (it was fine in Webstorm)

Reply:
I think if you set the return type in `.map<T>()` it works: 
<img width="555" alt="image" src="https://user-images.githubusercontent.com/1219069/161825723-c1fedc06-a637-43e6-8556-c760b49393d2.png">

Reply:
tried that